common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -113,11 +113,13 @@ class NativeAuthMsalController : BaseNativeAuthController() {
                     parameters as SignInStartUsingPasswordCommandParameters,
                     mergedScopes)
 
-                return processSignInInitiateApiResult(
+                val result = processSignInInitiateApiResult(
                     initiateApiResult = initiateApiResult,
                     oAuth2Strategy = oAuth2Strategy,
                     parametersWithScopes = parametersWithScopes,
                     usePassword = true)
+                StringUtil.overwriteWithZero(parametersWithScopes.password)
+                return result
```

Comment:
Alternatively you can choose to return from the method as you were and add a `finally` block corresponding to the try block and wipe the password in that finally block. Either way is fine.

Reply:
I have used try finally